torch/distributed/_composable/contract.py

```diff
@@ -196,5 +189,9 @@ def _get_registry(module: nn.Module) -> Dict[str, RegistryItem]:
     Get an ``OrderedDict`` of composable APIs that have been applied to the
     ``module``, indexed by the API name.
     """
-    default_registry: Dict[str, RegistryItem] = OrderedDict()
-    return module.__dict__.setdefault(REGISTRY_KEY, default_registry)  # type: ignore[call-overload]
+    if hasattr(module, REGISTRY_KEY):
+        return getattr(module, REGISTRY_KEY)
+    else:
+        default_registry: Dict[str, RegistryItem] = OrderedDict()
+        setattr(module, REGISTRY_KEY, default_registry)
```

Comment:
is setattr here always going to write, while ` module.__dict__.setdefault` would have read the existing value if existing?

Reply:
I thought the hasattr would protect it? Like, how would it overwrite an existing value here if we are below hasattr?

Reply:
well.. logically probably yes. but it is a legit difference and if module implements hasattr/getattr/setattr weirdly its possible something doesn't line up.

I couldn't see anything else wrong above. i would start ablating at this point to narrow it down

Reply:
I'll put it back and work around it for now, ty